sei-tendermint/libs/utils/canonical_test.go

```diff
@@ -0,0 +1,159 @@
+package utils
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/stretchr/testify/require"
+
+	testpb "github.com/tendermint/tendermint/proto_v2/test"
+)
+
+// Test checking that the canonical encoding is a valid proto encoding and that it is stable.
+func TestMarshalCanonicalRoundTrip(t *testing.T) {
```

Comment:
would be nice to add some `testdata` proto files and run some tests against those, showcasing the use while at it.